pages/post/_id.vue

```diff
@@ -1,12 +1,57 @@
 <template>
   <div>
-    <p>{{ $route.params }}</p>
+    <p>ポストID: {{ postId }}</p>
+    <p>タイトル: {{ postTitle }}</p>
+    <p>ユーザー: {{ userName }}</p>
+    <p>内容: {{ postText }}</p>
+    <audio controls :src="audioUrl"></audio>
   </div>
 </template>
 
 <script>
+import firebase from '~/plugins/firebase'
+// import { mapActions, mapGetters } from 'vuex'
+
+const db = firebase.firestore()
+const postsCollection = db.collection('posts')
+
 export default {
+  data: () => ({
+    postId: '',
+    postTitle: '',
+    userName: '',
+    postText: '',
+    audioUrl: '',
+  }),
+  created() {
```

Comment:
むしろ created を消して asyncDataでいいかも
```suggestion
  async asyncData({ params }) {
    // URLパラメータを取得
    const id = params.id

    const { postId, postTitle, userName, postText, audioUrl } = await postsCollection.doc(id).get().then(doc => {
      return doc.data()
    })
    return { postId, postTitle, userName, postText, audioUrl }
  },

```